packages/runtime/src/lib/itxn.ts

```diff
@@ -202,10 +202,15 @@ const _getASAConfigAddr = (addr?: Uint8Array): string => {
 /* eslint-disable sonarjs/cognitive-complexity */
 export function parseEncodedTxnToExecParams (tx: EncTx,
   interpreter: Interpreter, line: number): types.ExecParams {
+  // signer is the contract
+  const appID = interpreter.runtime.ctx.tx.apid ?? 0;
+  const applicationAccount = getApplicationAddress(appID);
+
   // initial common fields
   const execParams: any = {
     sign: types.SignType.SecretKey,
-    fromAccount: _getRuntimeAccount(tx.snd, interpreter, line), // sender is the contract
+    fromAccount: { addr: applicationAccount, sk: Buffer.from([]) }, // signer is the contract
```

Comment:
```suggestion
    fromAccount: { addr: appAddress, sk: Buffer.from([]) }, // signer is the contract
```

Reply:
What if i rekey the contract to another account? In that case i would need to change the signer, ryt? Can i set `authAddr` in an inner transaction? Could you please check?

Reply:
> What if i rekey the contract to another account? In that case i would need to change the signer, ryt? Can i set `authAddr` in an inner transaction? Could you please check?

I tried to do it before, but I stuck with rekeyTo on the inner transaction. I think it is fine now on teal v6. I will check it.